internal/github/github_author.go

```diff
@@ -0,0 +1,20 @@
+package github
+
+import (
+	"fmt"
+
+	"github.com/nestoca/joy/api/v1alpha1"
+)
+
+func GetCommitGitHubAuthor(project *v1alpha1.Project, defaultOrganization, commitSHA string) (string, error) {
+	repository := project.Spec.Repository
+	if repository == "" {
+		repository = fmt.Sprintf("%s/%s", defaultOrganization, project.Name)
+	}
+
+	out, err := executeAndGetOutput(".", "api", "--jq", ".author.login", "repos/"+repository+"/commits/"+commitSHA)
```

Comment:
This means we'd need to make an API query per commit. Can we do a single query for the commit range instead?

https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits:~:text=expects%20the%20format-,BASE...HEAD,-.%20Both%20must%20be

Reply:
That's definitely possible, but would require a significant bit of code to be refactored. I would think of that as premature optimization and revisit later if it proves to be an issue.

Reply:
I just tested the latest version. Fixing the only-one-commit bug did make the PR creation substantially slower (~15 seconds). So slow I was wondering if the process hanged.

```
JOY_DEV_SKIP_VERSION_CHECK=1 go run ./cmd/joy release promote --no-prompt  --source qa --target demo products --dry-run
```

I think it's good enough for now (let's merge), but I disagree with it being a premature optimization.

For now can we add a log message explaining what joy is doing?

```
Fetching commits details from GitHub...
```